File: app/src/main/java/it/cammino/risuscito/database/entities/CustomList.java
package it.cammino.risuscito.database.entities;

import android.arch.persistence.room.Entity;

import java.sql.Date;

@Entity(primaryKeys = {"id", "position", "idCanto"})
public class CustomList {

  public int id;

  public int position;

  public int idCanto;

  public Date timestamp;
}

File: app/src/main/java/it/cammino/risuscito/database/entities/CustomList.kt
package it.cammino.risuscito.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import it.cammino.risuscito.utils.StringUtils

import java.sql.Date

@Entity(primaryKeys = ["id", "position", "idCanto"])
class CustomList {

    var id: Int = 0

    var position: Int = 0

    var idCanto: Int = 0

    //IMPORTANTE PER FAR FUNZIONARE L'AUTO-MIGRATION DI ROOM
    @ColumnInfo(defaultValue = StringUtils.EMPTY)
    var notaPosizione: String = StringUtils.EMPTY

    var timestamp: Date? = null
}
